src/lib/utils/project/GitProjectService.ts

```diff
@@ -661,26 +671,82 @@ export default class GitProjectService {
             address: AddressDriverClient.getUserAddress(m.accountId),
           },
         })),
-        dependencies: projectMetadata.data.splits.dependencies.map((d) =>
-          'source' in d
-            ? {
-                type: 'repo',
-                weight: d.weight,
-                account: {
-                  driver: 'repo',
-                  accountId: d.accountId,
-                },
-                source: d.source,
+        dependencies: await Promise.all(
+          projectMetadata.data.splits.dependencies.map(async (d) => {
```

Comment:
Not for now and it's a personal preference, but since this might be changing in the future I'd abstract the whole dependencies calculation to something else.
Probably a separate function for now, and I'd make the comments functions themselves.

Reply:
Yeah, I also had an urge to refactor here, but decided not to given all this will be moot once we have the new service